qa/lucene-index-compatibility/src/javaRestTest/java/org/elasticsearch/lucene/FullClusterRestartLuceneIndexCompatibilityIT.java

```diff
@@ -173,6 +169,70 @@ public void testIndexUpgrade() throws Exception {
         }
     }
 
+    /**
+     * Creates an index on N-2, closes it on N-1 (without marking it as read-only), then upgrades to N.
+     */
+    public void testClosedIndexUpgrade() throws Exception {
+        final String index = suffix("index");
+        final int numDocs = 2437;
+
+        if (isFullyUpgradedTo(VERSION_MINUS_2)) {
+            createIndex(
+                client(),
+                index,
+                Settings.builder()
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, randomInt(2))
+                    .put(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), true)
```

Comment:
I see this in other places in this class so is probably necessary - though I do not understand why? Does not seem to be randomized in `createIndex`?

I may have asked this before but do not recall the answer, sorry if so.